File: server/modules/git/worktree-remove.ts
import { GitError, runGit } from "./exec.js";

export function removeWorktree(
  repoPath: string,
  worktreePath: string,
  options?: { force?: boolean },
): void {
  const args = ["worktree", "remove"];
  if (options?.force) args.push("--force");
  args.push(worktreePath);
  runGit(repoPath, args);
}

File: server-go/internal/git/worktree_remove.go
package git

// RemoveWorktree removes a linked worktree from the repository.
func RemoveWorktree(repoPath, worktreePath string, force bool) error {
	args := []string{"worktree", "remove"}
	if force {
		args = append(args, "--force")
	}
	args = append(args, worktreePath)
	_, err := Run(repoPath, args)
	return err
}
